src/extensions/geo-export/mesh-exporter.ts

```diff
@@ -111,7 +111,34 @@ export abstract class MeshExporter<D extends RenderObjectExportData> implements
         return interpolated.array;
     }
 
-    protected abstract addMeshWithColors(inpit: AddMeshInput): void;
+    protected static getInstance(input: AddMeshInput, instanceIndex: number) {
+        const { mesh, meshes } = input;
+        let vertices: Float32Array;
+        let normals: Float32Array;
+        let indices: Uint32Array | undefined;
+        let groups: Float32Array | Uint8Array;
+        let vertexCount: number;
+        let drawCount: number;
+        if (mesh !== undefined) {
+            vertices = mesh.vertices;
+            normals = mesh.normals;
+            indices = mesh.indices;
+            groups = mesh.groups;
+            vertexCount = mesh.vertexCount;
+            drawCount = mesh.drawCount;
+        } else {
+            const mesh = meshes![instanceIndex];
+            vertices = mesh.vertexBuffer.ref.value;
+            normals = mesh.normalBuffer.ref.value;
+            indices = mesh.indexBuffer.ref.value;
+            groups = mesh.groupBuffer.ref.value;
+            vertexCount = mesh.vertexCount;
+            drawCount = mesh.triangleCount * 3;
+        }
+        return { vertices, normals, indices, groups, vertexCount, drawCount };
```

Comment:
This is a little nitpick, but directly returning the object might be better than assigning mutable values, e.g.:

```ts
return mesh 
  ? { ...mesh }
  : { vertices: ... };
```

Reply:
Thanks! Fixed.